contracts/main/bridge/AsterizmClientUpgradeableTransparency.sol

```diff
@@ -0,0 +1,511 @@
+
+    /// Set initializer event
+    /// @param _initializerAddress address  Initializer address
+    event SetInitializerEvent(address _initializerAddress);
+
+    /// Set external relay event
+    /// @param _externalRelayAddress address  External relay address
+    event SetExternalRelayEvent(address _externalRelayAddress);
+
+    /// Set fee token event
+    /// @param _feeTokenAddress address  Fee token address
+    event SetFeeTokenEvent(address _feeTokenAddress);
+
+    /// Set local chain id event
+    /// @param _localChainId uint64
+    event SetLocalChainIdEvent(uint64 _localChainId);
+
+    /// Initiate transfer event (for client server logic)
+    /// @param _dstChainId uint64  Destination chein ID
+    /// @param _dstAddress uint  Destination address
+    /// @param _txId uint  Transaction ID
+    /// @param _transferHash bytes32  Transfer hash
+    /// @param _payload bytes  Payload
+    event InitiateTransferEvent(uint64 _dstChainId, uint _dstAddress, uint _txId, bytes32 _transferHash, bytes _payload);
+
+    /// Payload receive event (for client server logic)
+    /// @param _srcChainId uint64  Source chain ID
+    /// @param _srcAddress uint  Source address
+    /// @param _txId uint  Transfer ID
+    /// @param _transferHash bytes32  Transaction hash
+    event PayloadReceivedEvent(uint64 _srcChainId, uint _srcAddress, uint _txId, bytes32 _transferHash);
+
+    /// Add sender event
+    /// @param _sender address  Sender address
+    event AddSenderEvent(address _sender);
+
+    /// Remove sender event
+    /// @param _sender address  Sender address
+    event RemoveSenderEvent(address _sender);
+
+    /// Add trusted address event
+    /// @param _chainId uint64  Chain ID
+    /// @param _address uint  Trusted address
+    event AddTrustedAddressEvent(uint64 _chainId, uint _address);
+
+    /// Remove trusted address event
+    /// @param _chainId uint64  Chain ID
+    /// @param _address uint  Trusted address
+    event RemoveTrustedAddressEvent(uint64 _chainId, uint _address);
+
+    /// Set notify transfer sending result event
+    /// @param _flag bool  Notify transfer sending result flag
+    event SetNotifyTransferSendingResultEvent(bool _flag);
+
+    /// Set disable hash validation flag event
+    /// @param _flag bool  Use force order flag
+    event SetDisableHashValidationEvent(bool _flag);
+
+    /// Resend Asterizm transfer event
+    /// @param _transferHash bytes32  Transfer hash
+    /// @param _feeAmount uint  Additional fee amount
+    event ResendAsterizmTransferEvent(bytes32 _transferHash, uint _feeAmount);
+
+    /// Transfer sending result notification event
+    /// @param _transferHash bytes32  Transfer hash
+    /// @param _statusCode uint8  Status code
+    event TransferSendingResultNotification(bytes32 indexed _transferHash, uint8 _statusCode);
+
+    struct AsterizmTransfer {
+        bool successReceive;
+        bool successExecute;
+    }
+
+    struct AsterizmChain {
+        bool exists;
```

Comment:
what is the point of `exists` field if we can use `trustedAddresses[_chainId].trustedAddress != address(0)` instead